api/facadeversions.go

```diff
@@ -3,142 +3,146 @@
-	"EntityWatcher":                2,
-	"EnvironUpgrader":              1,
-	"ExternalControllerUpdater":    1,
-	"FanConfigurer":                1,
-	"FilesystemAttachmentsWatcher": 2,
-	"Firewaller":                   7,
-	"FirewallRules":                1,
-	"HighAvailability":             2,
-	"HostKeyReporter":              1,
-	"ImageMetadata":                3,
-	"ImageMetadataManager":         1,
-	"InstanceMutater":              3,
-	"InstancePoller":               4,
-	"KeyManager":                   1,
-	"KeyUpdater":                   1,
-	"LeadershipService":            2,
-	"LifeFlag":                     1,
-	"LogForwarding":                1,
-	"Logger":                       1,
-	"MachineActions":               1,
-	"MachineManager":               10,
-	"MachineUndertaker":            1,
-	"Machiner":                     5,
-	"MeterStatus":                  2,
-	"MetricsAdder":                 2,
-	"MetricsDebug":                 2,
-	"MetricsManager":               1,
-	"MigrationFlag":                1,
-	"MigrationMaster":              3,
-	"MigrationMinion":              1,
-	"MigrationStatusWatcher":       1,
-	"MigrationTarget":              2,
-	"ModelConfig":                  3,
-	"ModelGeneration":              4,
-	"ModelManager":                 9,
-	"ModelSummaryWatcher":          1,
-	"ModelUpgrader":                1,
-	"NotifyWatcher":                1,
-	"OfferStatusWatcher":           1,
-	"Payloads":                     1,
-	"PayloadsHookContext":          1,
-	"Pinger":                       1,
-	"Provisioner":                  11,
-	"ProxyUpdater":                 2,
-	"RaftLease":                    2,
-	"Reboot":                       2,
-	"RelationStatusWatcher":        1,
-	"RelationUnitsWatcher":         1,
-	"RemoteRelations":              2,
-	"RemoteRelationWatcher":        1,
-	"Resources":                    3,
-	"ResourcesHookContext":         1,
-	"RetryStrategy":                1,
-	"SecretsTriggerWatcher":        1,
-	"SecretBackends":               1,
-	"SecretBackendsManager":        1,
-	"SecretBackendsRotateWatcher":  1,
-	"SecretsRevisionWatcher":       1,
-	"Secrets":                      1,
-	"SecretsManager":               2,
-	"SecretsDrain":                 1,
-	"Singular":                     2,
-	"Spaces":                       6,
-	"SSHClient":                    4,
-	"StatusHistory":                2,
-	"Storage":                      6,
-	"StorageProvisioner":           4,
-	"StringsWatcher":               1,
-	"Subnets":                      5,
-	"Undertaker":                   1,
-	"UnitAssigner":                 1,
-	"Uniter":                       18,
-	"Upgrader":                     1,
-	"UpgradeSeries":                3,
-	"UpgradeSteps":                 2,
-	"UserManager":                  3,
-	"VolumeAttachmentsWatcher":     2,
-	"VolumeAttachmentPlansWatcher": 1,
+// We no longer support facade versions at 0.
+var facadeVersions = map[string][]int{
```

Comment:
Depending on the lifetime of a major version, these could get quite large. A tuple of lower/upper might be worth considering.